doc/topics/packaging/testing.rst

```diff
@@ -0,0 +1,117 @@
+.. _pkging-testing:
+
+================
+Testing Packages
+================
+
+The Package Test Suite
+======================
+
+The salt repo provides a test suite for testing basic functionality of our packages at ``<repo-root>/pkg/tests/``.
+The install, upgrade, and downgrade tests can be run locally and are run on most PRs that are submitted against Salt.
+
+
+.. warning::
+
+    These tests are destructive to your system, as they install the built packages onto the
+    system. They may also install older versions in the case of upgrades or downgrades.
+    It is recommended to run these in an isolated system, preferrably a VM.
+
+Setup
+=====
+In order to run the package tests, the `relenv <https://github.com/saltstack/relative-environment-for-python>`_
+onedir and built pacakges need to be placed in the correct locations.
+
+All salt packages for the wanted testing version must be placed in ``<repo-root>/pkg/artifacts/``,
+and the onedir must be located under ``<repo-root>/artifacts/``. Additionally, to ensure complete parity with Salt's CI/CD
+suite, the ``nox`` virtual environment can be placed at ``<repo-root>/.nox/test-pkgs-onedir`` The following are a few ways this can be accomplished easily.
+
+Using ``tools``
+---------------
+There is preliminary support for setting up the package test suite in the ``tools`` command suite that is located under ``<repo-root>/tools/testsuite/``.
+This does require the Github CLI tool ``gh`` to be properly configured for interaction with the salt repo.
+
+#. Install the dependencies
+
+    .. code-block:: bash
+
+       pip install -r requirements/static/ci/py{python_version}/tools.txt
+
+#. Download and lay down the artifacts
+
+    The most common use case is to test the packages built on a CI/CD run for a given PR.
+
+    .. code-block:: bash
+
+        tools ts setup --platform {linux|darwin|windows} --slug <operating-system-slug> --pr <pr-number> --pkg
+
+    To see the possible options for each argument, and other ways to utilize this command, use the following.
+
+    .. code-block:: bash
+
+        tools ts setup -h
+
+
+Downloading Individually
+------------------------
+If the ``tools ts setup`` command doesn't quite work, the artifacts can be downloaded, unzipped, and placed in the correct locations manually.
+Most often, you will be testing packages built on a CI/CD run for a given PR. This guide will explain how to setup for running the package tests using those artifacts.
```

Comment:
**Nitpick (style, non-blocking):** Our style guide says to use the present tense rather than the future tense where possible: https://developers.google.com/style/tense

Also, it should be "set up" when it's a verb.

Rewrite: "Typically, you want to test packages built on a CI/CD run for a given PR. This guide explains how to set up for running the package tests using those artifacts."